tests/language/test_completion_refresher.py

```diff
@@ -75,3 +75,39 @@ def test_refresh_with_callbacks(self):
             self.refresher.refresh(callbacks)
             self.refresher._completer_thread.join()
             self.assertEqual(callbacks[0].call_count, 1)
+            
+    def test_refresh_selects_pg_completer(self):
+        """
+        The correct completer (pg vs mysql) should be selected.
+        """
+        callbacks = Mock()
+        pg_completer = Mock()
+        mysql_completer = Mock()
+
+        mock_COMPLETER_MAP = {
+            PG_PROVIDER_NAME: pg_completer,
+            MYSQL_PROVIDER_NAME: mysql_completer
+        }
+        with patch('ossdbtoolsservice.language.completion_refresher.COMPLETER_MAP', mock_COMPLETER_MAP):
+            # Set refreshers to 0: we're not testing refresh logic here
+            self.refresher.refreshers = {}
+            self.refresher.refresh(callbacks)
+            pg_completer.assert_called_once()
```

Comment:
I am not sure I get this, but why is PG completer called and not MySQL in this test?

Reply:
PGCompleter, not MySQLCompleter, should be called because MockServerConnection inherits from PostgreSQLConnection, so its provider name is PG_PROVIDER_NAME. I can add a comment explaining this here and in the mysql completer test.